### DISCUSSION TITLE
Components with different onChange signature

### DISCUSSION THREAD

**Discussion by codinginflow** [react-hook-form/react-hook-form#10061] [Q&A]
How can I register an input component whose onChange callback looks like this:
![image](https://user-images.githubusercontent.com/52977034/223706382-ce414448-c28a-4b37-a067-ef7f34c53fcd.png)

I can call the `onChange` function of the register return directly, but I have no clue what to pass to it since the type of its argument is `event: { target: any; type?: any; }`.

![image](https://user-images.githubusercontent.com/52977034/223749717-282b2c1b-112e-4439-abff-338b699880a1.png)

Do I have to use `setValue` here?

**Comment by leapful** [comment]
@codinginflow 

The event handler `onChange` accepts object like `{ target: { value: 'YOUR_VALUE_FROM_EDITOR' }, type: 'text' }`. The value can be `text` string or `html` string depends on what data you expected to process.

**Reply by codinginflow** [reply]
I tried it, but RHF still doesn't detect the input. I'm using `setValue` right now.
